Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -179,409 +135,363 @@ public void Dispose()
+        /// </summary>
+        private void EnforceBindSettings(NetworkConfiguration config)
         {
-            port = null;
-            ArgumentNullException.ThrowIfNull(source);
-
-            // Do we have a source?
-            bool haveSource = !source.Address.Equals(IPAddress.None);
-            bool isExternal = false;
-
-            if (haveSource)
+            lock (_initLock)
             {
-                if (!IsIP6Enabled && source.AddressFamily == AddressFamily.InterNetworkV6)
+                // Respect explicit bind addresses
+                var localNetworkAddresses = config.LocalNetworkAddresses;
+                if (localNetworkAddresses.Length > 0 && !string.IsNullOrWhiteSpace(localNetworkAddresses.First()))
                 {
-                    _logger.LogWarning("IPv6 is disabled in Jellyfin, but enabled in the OS. This may affect how the interface is selected.");
+                    var bindAddresses = localNetworkAddresses.Select(p => IPAddress.TryParse(p, out var addresses)
+                        ? addresses
+                        : (_interfaces.Where(x => x.Name.Equals(p, StringComparison.OrdinalIgnoreCase))
+                            .Select(x => x.Address)
+                            .FirstOrDefault() ?? IPAddress.None))
+                        .ToList();
+                    bindAddresses.RemoveAll(x => x == IPAddress.None);
+                    _interfaces = _interfaces.Where(x => bindAddresses.Contains(x.Address)).ToList();
+
+                    if (bindAddresses.Contains(IPAddress.Loopback))
+                    {
+                        _interfaces.Add(new IPData(IPAddress.Loopback, new IPNetwork(IPAddress.Loopback, 8), "lo"));
+                    }
+
+                    if (bindAddresses.Contains(IPAddress.IPv6Loopback))
+                    {
+                        _interfaces.Add(new IPData(IPAddress.IPv6Loopback, new IPNetwork(IPAddress.IPv6Loopback, 128), "lo"));
+                    }
                 }
 
-                if (!IsIP4Enabled && source.AddressFamily == AddressFamily.InterNetwork)
+                // Remove all interfaces matching any virtual machine interface prefix
+                if (config.IgnoreVirtualInterfaces)
                 {
-                    _logger.LogWarning("IPv4 is disabled in Jellyfin, but enabled in the OS. This may affect how the interface is selected.");
-                }
+                    // Remove potentially exisiting * and split config string into prefixes
+                    var virtualInterfacePrefixes = config.VirtualInterfaceNames
+                        .Select(i => i.ToLowerInvariant()
+                            .Replace("*", string.Empty, StringComparison.OrdinalIgnoreCase));
 
-                isExternal = !IsInLocalNetwork(source);
+                    // Check all interfaces for matches against the prefixes and remove them
+                    if (_interfaces.Count > 0 && virtualInterfacePrefixes.Any())
```

Comment:
You're enumerating it twice. Just skip the length check.

```suggestion
                    if (_interfaces.Count > 0)
```